DeepONet Project Update: Model Accessibility and Checkpoint Restoration Fix

parsa-rahbari-82 opened the pull request:
### Description:  
This pull request provides a series of important updates to the DeepONet project, primarily focusing on improving model accessibility and bug fixes. The following changes have been made:

1. Command-Line Model Parameters:  
   - Added command-line arguments to give users more control over the model configuration without needing to modify the source code directly.  
   - Users can now adjust critical parameters such as learning rate, number of sensors, and training epochs via the command line using Python's argparse library.  
   
   Example usage:  
   
   `python deeponet_pde.py -p ode -m 200 --num-train 10000 --num-test 3000--lr 1e-3 --epochs 50000 --nn deeponet --activation relu --init Glorot --stacked False`
   

2. Checkpoint Restoration Bug Fix:  
   - Resolved a bug related to restoring checkpoints, which was caused by an incorrect file path.  
   - The model can now properly restore from saved states for both training and inference.

3. Legendre Transform Flexibility:  
   - Replaced the hardcoded time value (T=2) with a dynamic time variable.  
   - This update ensures that the model can work with any time interval, providing flexibility and preventing extrapolation errors.

4. Compatibility with DeepONet’s Dynamic Systems:  
   - Updated the structure of X_train and ensured proper formatting for X_test to be compatible with TensorFlow's compat.v1 backend.  
   - Added necessary changes to avoid type mismatch errors, such as casting arrays in X_test to np.float32.

   Example code snippet:  
   
`   X_test = tuple([arr.astype(dtype=np.float32) for arr in X_test])
`   

### Testing:  
These changes have been thoroughly tested and confirmed to work reliably with TensorFlow's compat.v1 backend.

MIND Lab:
- Supervisor: Dr. Mostafa Abbaszadeh
Team:
   - AmirAli Fakhari Zavareh
   - Melika Taheri Soufi
   - Parsa Rahbari

parsa-rahbari-82 committed:
compatibility bug fixed.

parsa-rahbari-82 committed:
tensorflow_intel removed from dependencies.

parsa-rahbari-82 committed:
dynamic module import for src

parsa-rahbari-82 committed:
fix bug for dynamic import.

parsa-rahbari-82 committed:
Revert "dynamic module import for src"

This reverts commit 38260b97ed1a4a9f384dc1c0f876616b79b5bcf7.

parsa-rahbari-82 committed:
tendorflow compatibility fixed.

parsa-rahbari-82 committed:
setting up river pollutants example.

parsa-rahbari-82 committed:
smaller model for faster training

parsa-rahbari-82 committed:
basic commands added.

parsa-rahbari-82 committed:
necessary commands added.

parsa-rahbari-82 committed:
redundant prints removed.

parsa-rahbari-82 committed:
Minor tensorflow 2 compatibility solved.

AmirAliFakhari committed:
major bugs for checkpoints solved

parsa-rahbari-82 committed:
Legendre transform and deeponet compatibility bugs fixed.

lululxvi commented on .gitignore:
```diff
@@ -0,0 +1,181 @@
+### Python ###
```
Don't upload this file.

lululxvi commented on src/deeponet_pde.py:
```diff
@@ -1,16 +1,15 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
+from __future__ import absolute_import, division, print_function
```
This import line is not needed anymore for python 3.

lululxvi commented on src/system.py:
```diff
@@ -1,36 +1,34 @@
-from __future__ import print_function
+from __future__ import absolute_import, division, print_function
 
+import config
```
Use the original order of import, i.e., import python libraries first, and then self-defined files.

parsa-rahbari-82 committed:
Stop tracking .gitignore

parsa-rahbari-82 committed:
import order resolved. redundant python2 imports removed.

parsa-rahbari-82 commented:
Dear Dr. Lu 
Thanks for the review! I’ve addressed the points you raised:

- Redundant python 2 removed.
- .gitignore file removed.
- Using original order of import.

lululxvi commented on src/spaces.py:
```diff
@@ -1,15 +1,11 @@
 
-import config
 from utils import eig
+import config
```
`import ...` first, and then `from ... import ...`

parsa-rahbari-82 committed:
Import ordering fixed.

parsa-rahbari-82 commented:
I reordered imports according to PEP 8 guidelines.

parsa-rahbari-82 committed:
fixing default advd example.

parsa-rahbari-82 committed:
including test folder

parsa-rahbari-82 committed:
including test folder for demo.

lululxvi commented on requirements.txt:
Why revise this file?

lululxvi commented on .gitignore:
No need to add this file. Different users may have different style.

lululxvi commented on src/test/u_xsin2x.dat:
What is this file?

parsa-rahbari-82 committed:
Redundant files removed.
- .gitignore
- requirements.txt

parsa-rahbari-82 committed:
Remove test file.

lululxvi commented on requirements.txt:
```diff
@@ -1,8 +0,0 @@
-deepxde
```
Keep this file as the original one.

lululxvi replied:
@AmirAliFakhari It seems the file roolls back to the wrong version.

AmirAliFakhari replied:
Dear Mr. Lu Lu,

I hope you're doing well. I've been working on the DeepONet project for the past few months and truly appreciate your feedback throughout the process.

To help streamline the next steps and avoid back-and-forth revisions, would it be possible for you to share a complete list of the issues or changes you'd like to see in the current version? This would really help me address everything thoroughly and ensure that the final result meets your expectations.

AmirAliFakhari committed:
Rolling back previous requirements.txt

lululxvi commented on src/deeponet_pde.py:
```diff
@@ -1,16 +1,13 @@
-
+import deepxde as dde
 import numpy as np
 import tensorflow as tf
```
```
import argparse
import deepxde as dde
import itertools
import numpy as np
import tensorflow as tf
```

lululxvi commented on src/deeponet_pde.py:
```diff
@@ -1,16 +1,13 @@
-from spaces import FinitePowerSeries, FiniteChebyshev, GRF
-from system import LTSystem, ODESystem, DRSystem, CVCSystem, ADVDSystem
-from utils import merge_values, trim_to_65535, mean_squared_error_outlier, safe_test
+from spaces import GRF, FiniteChebyshev, FinitePowerSeries
```
```
from spaces import FiniteChebyshev, FinitePowerSeries, GRF
```

lululxvi commented on src/deeponet_pde.py:
```diff
@@ -21,9 +18,10 @@ def test_u_lt(nn, system, T, m, model, data, u, fname):
         X_test = merge_values(X_test)
-    y_pred = model.predict(data.transform_inputs(X_test))
+    X_test = tuple([arr.astype(dtype=np.float32) for arr in X_test])
+    y_pred = model.predict(X_test)
```
Why no `data.transform_inputs`? This is for data normalization.

lululxvi commented on src/deeponet_pde.py:
```diff
@@ -103,11 +109,11 @@ def g(s, u, x):
         # Gravity pendulum
-        # k = 1
-        # return [s[1], - k * np.sin(s[0]) + u]
+        k = 1
```
There is `return` in line 108. The code here is not used.

lululxvi commented on src/spaces.py:
```diff
@@ -1,14 +1,10 @@
 from sklearn import gaussian_process as gp
-
-import config
 from utils import eig
```
```
import config

import matplotlib.pyplot as plt
...
```

lululxvi commented on src/system.py:
```diff
@@ -1,36 +1,34 @@
-from __future__ import print_function
-
 import numpy as np
+import config
```
`config` is a local file. Move it below as the original code.

lululxvi commented on src/utils.py:
```diff
@@ -1,12 +1,9 @@
-
 import sys
 import time
+import numpy as np
```
Use the original import format for `numpy`
